src/usdb_syncer/settings.py

```diff
@@ -487,6 +494,96 @@ def __str__(self) -> str:
+        match self:
+            case SupportedApps.KAREDI:
+                return "Karedi"
+            case SupportedApps.PERFORMOUS:
+                return "Performous"
+            case SupportedApps.ULTRASTAR_MANAGER:
+                return "UltraStar Manager"
+            case SupportedApps.USDX:
+                return "UltraStar Deluxe"
+            case SupportedApps.VOCALUXE:
+                return "Vocaluxe"
+            case SupportedApps.YASS_RELOADED:
+                return "YASS Reloaded"
+            case _ as unreachable:
+                assert_never(unreachable)
+
+    def executable_name(self) -> str:
+        match self:
+            case SupportedApps.KAREDI:
+                return "Karedi"
+            case SupportedApps.PERFORMOUS:
+                return "performous"
+            case SupportedApps.ULTRASTAR_MANAGER:
+                return "UltraStar-Manager"
+            case SupportedApps.USDX:
+                return "ultrastardx"
+            case SupportedApps.VOCALUXE:
+                return "Vocaluxe"
+            case SupportedApps.YASS_RELOADED:
+                return "yass"
+            case _ as unreachable:
+                assert_never(unreachable)
+
+    def songpath_parameter(self) -> str:
+        match self:
+            case SupportedApps.KAREDI:
+                return ""
+            case SupportedApps.PERFORMOUS:
+                return ""
+            case SupportedApps.ULTRASTAR_MANAGER:
+                return "-songpath"
+            case SupportedApps.USDX:
+                return "-songpath"
+            case SupportedApps.VOCALUXE:
+                return "-SongFolder"
+            case SupportedApps.YASS_RELOADED:
+                return ""
+            case _ as unreachable:
+                assert_never(unreachable)
+
+    def open_app(self, path: Path) -> None:
+        _logger.debug(f"Starting {self} with '{path}'.")
+        executable = get_app_path(self)
+        if executable.endswith(".jar"):
+            cmd = ["java", "-jar", executable, str(path)]
+        elif executable.endswith(".app"):
+            cmd = [
+                os.path.join(executable, "Contents", "MacOS", self.executable_name()),
+                self.songpath_parameter(),
+                str(path),
+            ]
+        else:
+            cmd = [executable, self.songpath_parameter(), str(path)]
+        try:
+            subprocess.Popen(cmd)  # pylint: disable=consider-using-with
+        except FileNotFoundError as fnf_error:
+            _logger.error(
+                f"File not found: {fnf_error}. "
+                "Please check the executable path in the settings."
+            )
+            traceback.print_exc()
+        except OSError as os_error:
+            _logger.error(f"OS error: {os_error}")
+            traceback.print_exc()
+        except subprocess.SubprocessError as sp_error:
+            _logger.error(f"Subprocess error: {sp_error}")
+            traceback.print_exc()
```

Comment:
The explicit error handling is very nice.
However, `print` statements won't show up in the log, and we usually give the user a more friendly message than the raw exception, like your second line for `FileNotFoundError`.
For the other cases, you can search the code base for `format_exc` for examples.
Also, the path that caused the issue would be helpful.

Reply:
So just a friendly message, and ignore the raw error? Or error with a friendly message and log raw exception as debug (in addition to traceback.format_exc)? Or does format_exc include the raw exception already?

Reply:
`format_exc` gives you the traceback including the exception. In case the file wasn't found, we know exactly what went wrong and this won't be very interesting. But in case of more ambiguous or unexpected errors, logging any information we can will be helpful when debugging.